usr/libexec/greenboot/greenboot

```diff
@@ -7,32 +7,52 @@ SCRIPTS_CHECK_PATHS=("/usr/lib/greenboot/check" "/etc/greenboot/check")
 SCRIPTS_GREEN_PATHS=("/usr/lib/greenboot/green.d" "/etc/greenboot/green.d")
 SCRIPTS_RED_PATHS=("/usr/lib/greenboot/red.d" "/etc/greenboot/red.d")
 
+source_configuration_file() {
+    GREENBOOT_CONFIGURATION_FILE=/etc/greenboot/greenboot.conf
+    if test -f "$GREENBOOT_CONFIGURATION_FILE"; then
+	# shellcheck source=/etc/greenboot/greenboot.conf
+        source $GREENBOOT_CONFIGURATION_FILE
+    fi
+}
+
+source_configuration_file
+function is_disabled {
+    HEALTHCHECK=$1
+    echo "${DISABLED_HEALTHCHECKS,,}" | tr " " '\n' | grep -F -q -x "$HEALTHCHECK"
+}
+
 script_runner () {
   local scripts_dir=$1; shift
   local mode=$1; shift
   local start_msg=$1; shift
   local required_hc_failed=false
   echo "$start_msg"
   for script in $(find "$scripts_dir" -name '*.sh' | sort); do
-    local rc=0
-    systemd-cat -t "$(basename "$script")" bash "$script" || rc=$?
-    if [ $rc -ne 0 ]; then
-      local failure_msg
-      failure_msg="Script '$(basename "$script")' FAILURE (exit code '$rc')"
-      case "$mode" in
-        "relaxed")
-          echo "<2>$failure_msg. Continuing..." >&2
-          ;;
-        "strict")
-          required_hc_failed=true
-          echo "<0>$failure_msg. Continuing..." >&2
-      esac
+    if is_disabled "$(basename "$script")"; then
+      echo "'$(basename "$script")' was skipped, as specified in config"
     else
-      echo "<6>Script '$(basename "$script")' SUCCESS"
+      local rc=0
+      systemd-cat -t "$(basename "$script")" bash "$script" || rc=$?
+      if [ $rc -ne 0 ]; then
+        local failure_msg
+        failure_msg="Script '$(basename "$script")' FAILURE (exit code '$rc')"
+        case "$mode" in
+          "relaxed")
+            echo "<2>$failure_msg. Continuing..." >&2
+            ;;
+          "strict")
+            required_hc_failed=true
+            echo "<0>$failure_msg. Continuing..." >&2
+        esac
+      else
+        echo "<6>Script '$(basename "$script")' SUCCESS"
+      fi
     fi
   done
 
-  [[ $required_hc_failed == false ]]
+  if [[ $required_hc_failed == true ]]; then
+    exit 1
+  fi
```

Comment:
This is almost equivalent but instead of returning an error the function exits the entire program with error, why do you need to change this behavior?

Reply:
This was not an intentional change, and I'm not sure why it did. It seems that when if I fail a reboot after making changes and unlocking with `ostree admin unlock --hotfix` that change is already present, even though I never wrote it. Investigating more when I get a chance

Reply:
Addressed and reverted in the most recent changes. Resolving this conversation